builder/penv_setup.py

```
@@ -806,6 +806,57 @@ def _install_esptool_from_tl_install(platform, python_exe, uv_executable, uv_cac
+            capture_output=True,
+            text=True,
+            timeout=5
+        )
+        if result.returncode == 0:
+            return
+    except Exception:
+        pass
+
+    # Install pio-lock from GitHub repository
+    try:
+        subprocess.check_call([
```

Ruff (0.15.12)

[error] `subprocess` call: check for execution of untrusted input

(S603)